baseband/gsb/payload.py

```diff
@@ -76,6 +77,9 @@ class GSBPayload(VLBIPayloadBase):
                  8: decode_8bit}
     _dtype_word = np.int8
 
+    _sample_shape_cls_1thread = namedtuple('SampleShape', 'nchan')
+    _sample_shape_cls_nthread = namedtuple('SampleShape', 'nthread, nchan')
+
```

Comment:
I think it might make it possible to make the rest more elegant by writing `_sample_shape_cls` as a function:
```
@staticmethod
def _sample_shape_cls(*args):
    if len(args) == 1:
        return _sample_shape_cls_1thread(args[0])
    else:
        return _sample_shape_cls_nthread(*args)
```
I think with that `fromdata` no longer has to be overridden...

Reply:
So it does!  I've updated `GSBPayload` and added a few lines to `test_gsb.py` to ensure we're still getting the correct sample shapes.

Made corresponding changes below.